c2rust-analyze/src/c_void_casts.rs

```diff
@@ -320,41 +405,44 @@ impl<'tcx> CVoidCasts<'tcx> {
             };
             let func_ty = func.ty(&body.local_decls, tcx);
 
-            // For [`CVoidCastDirection::From`], we only count
-            // a cast from `*c_void` to an arbitrary type in the subsequent block,
-            // searching forward.
-            let find_first_cast_succ_block = |get_cast| {
-                let successor_block_id = target.unwrap();
-                body.basic_blocks()[successor_block_id]
-                    .statements
-                    .iter()
-                    .find_map(get_cast)
-            };
-
-            // For [`CVoidCastDirection::From`], we only count
-            // a cast to `*c_void` from an arbitrary type in the same block,
-            // searching backwards.
-            let find_last_cast_curr_block =
-                |get_cast| bb_data.statements.iter().rev().find_map(get_cast);
-
             for direction in CVoidCastDirection::from_callee(ty_callee(tcx, func_ty))
                 .iter()
                 .copied()
             {
                 use CVoidCastDirection::*;
-                let c_void_ptr = match direction {
+                let c_void_ptr_place = match direction {
                     From => destination,
                     To => args[0].place().unwrap(),
                 };
-                let c_void_ptr = CVoidPtr::checked(c_void_ptr, &body.local_decls, tcx);
+                let c_void_ptr = CVoidPtr::checked(c_void_ptr_place, &body.local_decls, tcx);
                 let get_cast =
-                    move |stmt: &Statement<'tcx>| c_void_ptr.get_cast_from_stmt(direction, stmt);
+                    |stmt: &Statement<'tcx>| c_void_ptr.get_cast_from_stmt(direction, stmt);
```

Comment:
Since this is the only callback ever passed to `find_*_cast_*_block`, would it make sense to pass `c_void_ptr` and `stmt` to those functions instead, and inline this `get_cast_from_stmt` call?

Reply:
I don't mind either way, I'll play around with it to see if one appears to be better than the other.

Reply:
I only added the closure initially to avoid passing `direction` and making the code a bit shorter.  You can switch to whatever's simpler now.

Reply:
b6e492d4